Vote 結果の理由に Personality を表示

runceel opened the pull request:
Vote 結果の理由に Personality を表示するように変更しました。
<img width="2289" height="1315" alt="image" src="https://github.com/user-attachments/assets/46037630-e51c-47d0-a658-8c1f9830f76d" />

runceel committed:
投票理由の構造を改善し、セキュリティを強化

`MagiResponse` と `SageResponse` の構造を変更し、投票者のパーソナリティを含む `VoteReason` を導入しました。
`MagiService` での投票理由の生成方法を更新し、AI応答の管理を強化しました。
`MagiVote.razor` での表示部分も更新され、投票理由の詳細が反映されるようになりました。
さらに、プロジェクトファイルにユーザーシークレットの ID を追加し、セキュリティを向上させました。

07JP27 commented on MagiSystem.Core/VoteReason.cs:
```diff
@@ -0,0 +1,2 @@
+namespace MagiSystem.Core;
+public record VoteReason(string Personality, string ReasonText);
```
これをVoteResultとかにして、Vote結果も入れたほうが取り回しがいいかも？

07JP27 replied:
そうなるとプロパティがSageResponseと同じなので、そのままSageResponseのリストで返してもいいかもですね。

runceel replied:
SagaResponse を返すようにしました。
YesReasons と NoReasons プロパティの型が SagaResponse というのに違和感を感じたので YesResponses と NoResponses にリネームしてみたんですけどどうですかね？

07JP27 replied:
LGTM!

07JP27 committed:
Update Program.cs

runceel committed:
MagiResponseのプロパティを更新

MagiResponseクラスのYesReasonsとNoReasonsをそれぞれYesResponsesとNoResponsesに変更しました。これに伴い、DisplayResultメソッドやMagiServiceクラスのインスタンス生成部分も修正しました。また、VoteReasonクラスを削除し、YesResponsesとNoResponsesはSageResponse型のリストとして扱うようになりました。Razorコンポーネント内の条件分岐やループも更新され、表示内容が改善されました。

runceel committed:
Merge branch 'main' of https://github.com/runceel/MagiSystem

07JP27 committed:
Update Program.cs
